Dockerfile

```diff
@@ -1,7 +1,7 @@
 # syntax=docker/dockerfile:1
 
-ARG CUDA_VER=12.0.1
-ARG PYTHON_VER=3.11
+ARG CUDA_VER=unset
+ARG PYTHON_VER=unset
```

Comment:
Why has the default been removed here?

Reply:
So that we can be more confident that the values defined in the CI matrix are actually being passed through as docker builds args. If they're not passed through correctly, we'll now get loud errors near the source of the issue, instead of silently pulling an unintended old version as a result of falling back to defaults.

For more context, I did something similar here: https://github.com/rapidsai/ci-imgs/pull/102

I found these lines doing this:

```shell
git grep -E '12\.'
```

To be sure I added CUDA 12.5 correctly everywhere it needed to be added.

Now that the next time we do one of these migrations, no one will find these lines or have to worry about them.